daprdocs/content/en/developing-applications/building-blocks/workflow/howto-manage-workflow.md

```diff
@@ -35,6 +35,72 @@ await daprClient.TerminateWorkflowAsync(instanceId, workflowComponent);
 
 {{% /codetab %}}
 
+<!--Python-->
+{{% codetab %}}
+
+Manage your workflow within your code. In the `OrderProcessingWorkflow` example from the [Author a workflow]({{< ref "howto-author-workflow.md#write-the-workflow" >}}) guide, the workflow is registered in the code using the following APIs:
+- **start_workflow**: Start an instance of a workflow
+- **get_workflow**: Get information on a single workflow
+- **terminate_workflow**: Terminate or stop a particular instance of a workflow
+- **raise_event**: Raise an event on a workflow
+- **pause_workflow**: Pauses or suspends a workflow instance that can later be resumed
+- **resume_workflow**: Resumes a paused workflow instance
+- **purge_workflow**: Removes all metadata related to a specific workflow instance
+
+```python
+dapr = DaprGrpcClient(f'localhost:{self.server_port}')
+
+# Sane parameters
+workflow_name = 'testWorkflow'
+instance_id = str(uuid.uuid4())
+workflow_component = "dapr"
+input = "paperclips"
+
+# Start the workflow
+start_response = dapr.start_workflow(instance_id, workflow_name, workflow_component,
+                                     input.encode('utf-8'), None)
+self.assertEqual(instance_id, start_response.instance_id)
+
+
+
+# Get info on the workflow to check that it is running
+get_response = dapr.get_workflow(instance_id, workflow_component)
+self.assertEqual("RUNNING", get_response.runtime_status)
+
+# Pause the workflow
+dapr.pause_workflow(instance_id, workflow_component)
+
+# Get info on the workflow to check that it is paused
+get_response = dapr.get_workflow(instance_id, workflow_component)
+self.assertEqual("SUSPENDED", get_response.runtime_status)
+
+# Resume the workflow
+dapr.resume_workflow(instance_id, workflow_component)
+
+# Get info on the workflow to check that it is resumed
+get_response = dapr.get_workflow(instance_id, workflow_component)
+self.assertEqual("RUNNING", get_response.runtime_status)
+
+# Raise an event on the workflow. 
+TODO: Figure out how to check/verify this
```

Comment:
There is a TODO in the code here

Reply:
This doc was grabbed from the following PR: https://github.com/dapr/python-sdk/pull/554 This TODO and the one below were grabbed from the workflow.py example which was still being polished at the time of the creation of this document. Since the example in the PR was/is still being finalized, this portion included the TODOs. Once the example is fully completed, this file will reflect the finalized version.